examples/step-7/step-7.cc

```diff
@@ -998,19 +997,16 @@ namespace Step7
             GridGenerator::hyper_cube(triangulation, -1, 1);
             triangulation.refine_global(3);
 
-            typename Triangulation<dim>::cell_iterator cell =
-                                                         triangulation.begin(),
-                                                       endc =
-                                                         triangulation.end();
-            for (; cell != endc; ++cell)
+            for (const auto cell : triangulation.cell_iterators())
```

Comment:
Would you mind using `const auto& cell` here?

Reply:
Well "cell" is an iterator and `cell_iterators` and related return an iterator range. So using `auto &cell`, or `auto cell` doesn't make a difference here. I don't mind changing it to `const auto &cell` but I guess @bangerth has some strong opinions on that matter. (He asked me to replace a `const auto &center = ...` by `const auto center = ...` because the right hand side returned a value not a refererence.)

Reply:
I see that we used `const auto &cell` everywhere in the examples. So let me unify it.

Reply:
Yes, I really don't have much of an opinion in this context.